api_tests/scripts/generic_request.py

```diff
@@ -116,7 +116,8 @@ def check_status_code(self, response: 'response type', expected_status_code: int
         return response.status_code == expected_status_code
 
     def check_endpoint(self, verb: str, endpoint: str, expected_status_code: int,
-                       expected_response: dict or str or list, **kwargs) -> bool:
+                       expected_response: dict or str or list, return_response: bool = False,
+                       **kwargs) -> bool or object:
```

Comment:
I don't think `bool or object` is actually valid here, it should be `Union[bool, object]`.

That said as well, `object` is a bit too broad of a return type.

I think this function is probably doing too much and needs to be split in two, OR just return the response and do the bool conversion (checking truthiness) a level up

Reply:
Or any if the other `type or othertype` bits for that matter.